sorts/merge_sort_fastest.py

```diff
@@ -4,16 +4,51 @@
 Best Case Scenario : O(n)
 Worst Case Scenario : O(n)
 '''
+# These modules are required for testing only.
+import random, copy
+import matplotlib.pyplot as plt
+
 def merge_sort(LIST):
+    '''
+    Returns sorted list.
+    '''
+    # Divide list into two parts
     start = []
     end = []
-    while len(LIST) > 1:
-        a = min(LIST)
-        b = max(LIST)
+    a = min(LIST)
+    b = max(LIST)
+    
+    while len(LIST)>1:
         start.append(a)
         end.append(b)
-        LIST.remove(a)
-        LIST.remove(b)
-    if LIST: start.append(LIST[0])
+        try:
+            LIST.remove(a)
+            LIST.remove(b)
+            a = min(LIST)
+            b = max(LIST)
+        except ValueError:
+            continue
     end.reverse()
-    return (start + end)
+    return start + end
+
+def visualize(LIST, SORTED_LIST):
+    '''
+    Plots unsorted and sorted data.
+    '''
+    plt.plot(LIST)
```

Comment:
Changed the code like this:
```python
figure1 = plt.figure(1)
plt.plot(LIST)
plt.xlabel('Random Data')
figure1.show()
figure2 = plt.figure(2)
plt.plot(SORTED_LIST)
plt.xlabel('Merge Sorted Data')
figure2.show()

raw_input() # keeps the figures alive
```

With the code above it show up two figures parallel at the same time.